pkg/crosscluster/logical/replication_statements.go

```diff
@@ -20,40 +20,47 @@ import (
 type columnSchema struct {
 	column       catalog.Column
 	isPrimaryKey bool
+	isComputed   bool
 }
 
-// getPhysicalColumns returns the list of columns that are part of the table's
-// primary key and value.
-func getPhysicalColumns(table catalog.TableDescriptor) []catalog.Column {
-	columns := table.AllColumns()
-	result := make([]catalog.Column, 0, len(columns))
-	for _, col := range columns {
-		if !col.IsComputed() && !col.IsVirtual() && !col.IsSystemColumn() {
-			result = append(result, col)
-		}
-	}
-	return result
-}
-
-func getPhysicalColumnsSchema(table catalog.TableDescriptor) []columnSchema {
-	columns := getPhysicalColumns(table)
-	primaryIdx := table.GetPrimaryIndex()
-
+// getColumnSchema returns the list of all columns that is decoded by the CRUD
+// writer. It returns columns in column key order. The crud writer passes
+// around a tree.Datums for each row where column[i] is the column definition
+// for datums[i].
+//
+// A column is decoded by crud LDR writer if either are true:
+//  1. The column is part of the primary key. Every primary key column is needed
+//     to perform read refreshes.
+//  2. The column is not a computed column. If a column is not computed, it must
+//     be included in update and insert statements.
```

Comment:
nit: it may be worth mentioning the columns that we do _not_ bother to decode:
- SystemColumns
- Stored computed columns not apart of the pk.

Reply:
Done